tests/testfile_bitbucket/expected_iam.yml

```diff
@@ -0,0 +1,138 @@
+          script:
+            - export SAM_TEMPLATE="template.yaml"
+            - export PERMISSIONS_PROVIDER="IAM"
+            - export PIPELINE_USER_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
+            - export PIPELINE_USER_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
+            - export TESTING_PIPELINE_EXECUTION_ROLE="test-pipeline-execution-role"
+            - export TESTING_CLOUDFORMATION_EXECUTION_ROLE="test-cfn-execution-role"
+            - export TESTING_ARTIFACTS_BUCKET="test-bucket"
+            - export TESTING_REGION="us-east-2"
+            - export TESTING_IMAGE_REPOSITORY="test-ecr"
+            - export PRODUCTION_PIPELINE_EXECUTION_ROLE="prod-pipeline-execution-role"
+            - export PRODUCTION_CLOUDFORMATION_EXECUTION_ROLE="prod-cfn-execution-role"
+            - export PRODUCTION_ARTIFACTS_BUCKET="prod-bucket"
+            - export PRODUCTION_REGION="us-west-2"
+            - export PRODUCTION_IMAGE_REPOSITORY="prod-ecr"
+            - sam build --template $SAM_TEMPLATE --use-container
+            - source assume-role.sh $TESTING_PIPELINE_EXECUTION_ROLE testing-stage-packaging testing-stage "$PERMISSIONS_PROVIDER" "$BITBUCKET_STEP_OIDC_TOKEN"
+            - >
+              sam package --profile testing-stage
+              --s3-bucket $TESTING_ARTIFACTS_BUCKET
+              --region $TESTING_REGION
+              --image-repository ${TESTING_IMAGE_REPOSITORY}
+              --output-template-file packaged-testing.yaml
+            - source assume-role.sh $PRODUCTION_PIPELINE_EXECUTION_ROLE testing-stage-packaging production-stage "$PERMISSIONS_PROVIDER" $BITBUCKET_STEP_OIDC_TOKEN
+            - >
+              sam package --profile production-stage
+              --s3-bucket $PRODUCTION_ARTIFACTS_BUCKET
+              --region $PRODUCTION_REGION
+              --image-repository ${PRODUCTION_IMAGE_REPOSITORY}
+              --output-template-file packaged-production.yaml
+          artifacts:
+            - packaged-testing.yaml
+            - packaged-production.yaml
+          services:
+            - docker
+      - step:
+          oidc: true
+          name: Deploy to Test
+          script:
+            - export SAM_TEMPLATE="template.yaml"
+            - export PERMISSIONS_PROVIDER="IAM"
+            - export PIPELINE_USER_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
+            - export PIPELINE_USER_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
+            - export TESTING_PIPELINE_EXECUTION_ROLE="test-pipeline-execution-role"
+            - export TESTING_CLOUDFORMATION_EXECUTION_ROLE="test-cfn-execution-role"
+            - export TESTING_ARTIFACTS_BUCKET="test-bucket"
+            - export TESTING_STACK_NAME="test-stack"
+            - export TESTING_REGION="us-east-2"
+            - export TESTING_IMAGE_REPOSITORY="test-ecr"
+            - source assume-role.sh $TESTING_PIPELINE_EXECUTION_ROLE testing-stage-packaging testing-stage "$PERMISSIONS_PROVIDER" $BITBUCKET_STEP_OIDC_TOKEN
```

Comment:
nit: Is there particular reason we mix the usage of `"${PERMISSIONS_PROVIDER}"` and `"$PERMISSIONS_PROVIDER"` in this PR (for consistency)?

Reply:
It was mostly to keep it consistent within the template since in the bitbucket template it doesn't wrap it with `{}` but in the gitlab one it does when calling the assume role script

Reply:
I did test all the templates with the systems before, but since they changed a fair bit im running through it all again